Multiplication Chart

varun13169 opened the pull request:
This activity is based on Montessori Multiplication Chart Activity:
http://www.infomontessori.com/mathematics/tables-of-arithmetic-multiplication-charts.htm

A child will be given a multiplicand and a multiplier, he/she needs to select row and column corresponding to either multiplicand or multiplier and then report back the answer.

Here is how it looks.
![layout](https://cloud.githubusercontent.com/assets/10177043/14009776/5075b3c0-f1b5-11e5-815b-370753f9700d.png)

petitlapin commented:
Hi,

you can probably center the texts in the cases, it should be better.

Johnny

petitlapin commented on src/activities/multiplication-chart/MultiplicationChart.qml:
```diff
@@ -0,0 +1,287 @@
+    pageComponent: Image {
+        id: background
+        fillMode: Image.PreserveAspectCrop
+        source: "./resource/numbers.svg"
```
you should use the qrc path ("qrc:/gcompris/src/activities/{activity}/resource/").

petitlapin commented on src/activities/multiplication-chart/MultiplicationChart.qml:
```diff
@@ -0,0 +1,287 @@
+            spacing: items.spacing
+            columns: 1
+            rows: 11
+            Repeater {
```
As it is the same code as above, you probably can do a separate item for it

petitlapin commented on src/activities/multiplication-chart/multiplication-chart.js:
```diff
@@ -0,0 +1,129 @@
+    items.rowSelected = 0
+    items.colSelected = 0
+    items.enteredInput = 0
+    items.answer.text = qsTr("")
```
no need of qsTr for empty strings

petitlapin commented on src/activities/multiplication-chart/multiplication-chart.js:
```diff
@@ -0,0 +1,129 @@
+
+
+function makeOtherColInRowWhite() {
+    for(var i=0; i<items.repeaterGridRow.count; i++) {
```
Can't the state be directly binded to the value of the selected row/column? I mean, in the Repeater having something like:
state: index == items.rowSelected ? "active" : "default"

petitlapin commented on src/activities/multiplication-chart/MultiplicationChart.qml:
```diff
@@ -0,0 +1,287 @@
+            text: items.rowQues + " X " + items.colQues + " = "
+        }
+
+        TextInput {
```
We have a Numpad item in core that could be better than having a TextInput, what do you think?

varun13169 commented:
Thanks for the suggestions.
Will look at them and will incorporate them in a week :)

varun13169 commented:
Hi,
1) I added numpad to this activity as was suggested, now it function similar to algebra_by.

2) Moved the answer in middle, changed instruction font to small.

3) I tried to remove the loop in JS that was changing the state but it didn’t work well, reason is when mouse is clicked index have the value of the current rectangle only, so no knowledge about past is there hence loop is needed.
If I keep it outside the mouse area then it works but only for first iteration at the time of initialisation but I need it to check whenever mouse is clicked.

Now activity looks like this.
![screenshot from 2016-03-28 00 45 15](https://cloud.githubusercontent.com/assets/10177043/14067352/2b0ed16a-f481-11e5-9b6c-a131a396b72d.png)
